spec/rspec/core/memoized_helpers_spec.rb

```diff
@@ -520,12 +520,39 @@ def count
       end.to raise_error(/#let or #subject called without a block/)
     end
 
-    it 'raises an error when attempting to define a reserved method name' do
-      expect do
-        RSpec.describe { let(:initialize) { true }}
-      end.to raise_error(/#let or #subject called with a reserved name #initialize/)
+    context 'when defining a reserved name' do
+      context 'when name is initialize' do
+
+        it 'raises an error when attempting to define a reserved name #initialize' do
+          expect do
+            RSpec.describe { let(:initialize) { true }}
+          end.to raise_error(/#let or #subject called with reserved name #initialize/)
+        end
+
+        it 'raises an error when attempting to define a reserved name #initialize as a string' do
+          expect do
+            RSpec.describe { let('initialize') { true }}
+          end.to raise_error(/#let or #subject called with reserved name #initialize/)
+        end
+      end
+
+      context 'when name is to_s' do
+        it 'raises an error when attempting to define a reserved name #to_s' do
+          expect do
+            RSpec.describe { let(:to_s) { true }}
+          end.to raise_error(/#let or #subject called with reserved name #to_s/)
+        end
+
+        it 'raises an error when attempting to define a reserved name #to_s as a string' do
+          expect do
+            RSpec.describe { let('to_s') { true }}
+          end.to raise_error(/#let or #subject called with reserved name #to_s/)
```

Comment:
I think that `#to_s` is typically used to refer to a method name, while when referring to an argument for `let`/`subject` it would be correct to wrap it in backticks.